packages/vega-interpreter/src/functions.js

```diff
@@ -7,8 +7,13 @@ const apply = (m, args, cast) => {
   return obj[m].apply(obj, slice.call(args, 1));
 };
 
-const datetime = (y, m, d, H, M, S, ms) =>
-  new Date(y, m || 0, d != null ? d : 1, H || 0, M || 0, S || 0, ms || 0);
+const datetime = (...args) => {
```

Comment:
This code is called many times, so we would benefit from not needing to create 2 intermediary arrays per call (one with the spread operator, the other with the args destructure).

What do you think of something like this?

```js
const datetime = (yearOrTimestring, m = 0, d = 1, H = 0, M = 0, S = 0, ms = 0) =>
  typeof yearOrTimestring === 'string' 
    ? new Date(yearOrTimestring)
    : new Date(yearOrTimestring, m, d, H, M, S, ms);
```

Reply:
This looks good, I will adopt this change.